test/test_log.dart

```diff
@@ -0,0 +1,15 @@
+import 'dart:async';
+
+import 'package:checks/checks.dart';
+import 'package:zulip/log.dart';
+
+Future<Subject<List<String>>> checkLogs(FutureOr<void> Function() callback) async {
+  assert(logHistory == null);
+  logHistory = [];
+  try {
+    await callback();
+    return check(logHistory).isA<List<String>>();
```

Comment:
```suggestion
    return check(logHistory!);
```

If `logHistory` is back to null at this point, that seems like not just a normal failure of the calling test case — it means something that shouldn't be messing with that variable has a surprising bug where it's doing so anyway. So we can avoid cluttering test failure messages with a line about how the log history was expected to be (and is) a `List<String>`.